src/pages/Swap/LimitOrderBox/components/OrderLimitPriceField/OrderLimitPriceField.tsx

```diff
@@ -0,0 +1,200 @@
+import { formatUnits, parseUnits } from '@ethersproject/units'
+import { Token, TokenAmount } from '@swapr/sdk'
+
+import { useContext, useEffect, useState } from 'react'
+import { RefreshCw } from 'react-feather'
+import { useTranslation } from 'react-i18next'
+
+import { invalidChars } from '../../constants'
+import { LimitOrderFormContext } from '../../contexts/LimitOrderFormContext'
+import { LimitOrderKind } from '../../interfaces'
+import { InputGroup } from '../InputGroup'
+import {
+  LimitLabel,
+  MarketPriceDiff,
+  SetToMarket,
+  SwapTokenIconWrapper,
+  SwapTokenWrapper,
+  ToggleCurrencyButton,
+} from './styles'
+
+export interface OrderLimitPriceFieldProps {
+  id?: string
+}
+
+export function OrderLimitPriceField({ id }: OrderLimitPriceFieldProps) {
+  const { t } = useTranslation('swap')
+  const {
+    limitOrder,
+    buyTokenAmount,
+    sellTokenAmount,
+    setLimitOrder,
+    formattedLimitPrice,
+    setFormattedLimitPrice,
+    setBuyTokenAmount,
+    setFormattedBuyAmount,
+    setToMarket,
+    marketPrices,
+  } = useContext(LimitOrderFormContext)
+
+  const [baseTokenAmount, quoteTokenAmount] =
+    limitOrder.kind === LimitOrderKind.SELL ? [sellTokenAmount, buyTokenAmount] : [buyTokenAmount, sellTokenAmount]
+  const inputGroupLabel = `${limitOrder.kind} ${baseTokenAmount?.currency?.symbol} at`
+  const toggleCurrencyButtonLabel = `${quoteTokenAmount?.currency?.symbol}`
```

Comment:
I would suggest putting this in place where its used since its used only once

Reply:
Some named variables don't harm, I think.